src/routes/categories/containers/CategoriesContainer.js

```diff
@@ -4,59 +4,63 @@ import { withTheme } from 'styled-components/native'
 import { connect } from 'react-redux'
 import compose from 'lodash/fp/compose'
 
-import type { CategoryRouteStateType, StateType } from '../../../modules/app/StateType'
+import type { StateType } from '../../../modules/app/StateType'
 import withRouteCleaner from '../../../modules/endpoint/hocs/withRouteCleaner'
 import { type Dispatch } from 'redux'
 import CategoriesRouteStateView from '../../../modules/app/CategoriesRouteStateView'
 import type { StoreActionType } from '../../../modules/app/StoreActionType'
 import createNavigateToCategory from '../../../modules/app/createNavigateToCategory'
-import { CityModel } from '@integreat-app/integreat-api-client'
 import createNavigateToIntegreatUrl from '../../../modules/app/createNavigateToIntegreatUrl'
 import { branch, renderComponent } from 'recompose'
 import CategoryLanguageNotAvailableContainer from './CategoryLanguageNotAvailableContainer'
 import CategoriesScrollView from '../components/CategoriesScrollView'
+import { Failure } from '../../../modules/error/components/Failure'
 
 const mapDispatchToProps = (dispatch: Dispatch<StoreActionType>, ownProps) => ({
   navigateToCategory: createNavigateToCategory('Categories', dispatch, ownProps.navigation),
   navigateToIntegreatUrl: createNavigateToIntegreatUrl(dispatch, ownProps.navigation)
 })
 
 const mapStateToProps = (state: StateType, ownProps) => {
-  const targetCityCode: CityModel = ownProps.navigation.getParam('cityCode')
-  const key: string = ownProps.navigation.getParam('key')
-
-  const targetRoute: CategoryRouteStateType = state.cityContent.categoriesRouteMapping[key]
-  const language = state.cityContent.language
-
-  if (!targetRoute || !language) {
-    return {
-      cityCode: targetCityCode,
-      language: language,
-      cities: state.cities.models
-    }
+  const {resourceCache, categoriesRouteMapping, languages, language, city} = state.cityContent
+
+  if (languages && !languages.map(languageModel => languageModel.code).includes(language)) {
+    return {invalidLanguage: true}
+  }
+
+  if (state.cities.errorMessage !== undefined ||
+    categoriesRouteMapping.errorMessage !== undefined ||
+    resourceCache.errorMessage !== undefined) {
```

Comment:
Alternative for check:

```
if ([state.cities.errorMessage, resourceCache.errorMessage, categoriesRouteMapping.errorMessage].includes(undefined)) {
```

Reply:
For some reasons both suggestions are producing new flow errors. I'll just merge it.